### ISSUE TITLE
Panic in an only advertise IPV6 routes mode 

### ISSUE THREAD

**Issue by davidjbrady** [cloudnativelabs/kube-router#601]
Howdy,

I've been experimenting with kube-router in an "**only advertise IPV6 routes mode**" and appear to have come across a bug. Consistently and a few moments after launching kube-router, it panics and outputs the following.

> panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe8ecdc]
> goroutine 131 [running]:
> github.com/cloudnativelabs/kube-router/vendor/github.com/coreos/go-iptables/iptables.(*IPTables).Exists(0x0, 0x193e028, 0x6, 0x193eeb8, 0x7, 0xc00061b968, 0x8, 0x8, 0x0, 0x0, ...)
> 	/src/github.com/cloudnativelabs/kube-router/vendor/github.com/coreos/go-iptables/iptables/iptables.go:92 +0x2c
> github.com/cloudnativelabs/kube-router/pkg/controllers/routing.(*NetworkRoutingController).enableForwarding(0xc0006c8a00, 0x0, 0x0)
> 	/src/github.com/cloudnativelabs/kube-router/pkg/controllers/routing/network_routes_controller.go:597 +0xdc
> github.com/cloudnativelabs/kube-router/pkg/controllers/routing.(*NetworkRoutingController).Run(0xc0006c8a00, 0xc000534ba0, 0xc000176600, 0xc00052bfc0)
> 	/src/github.com/cloudnativelabs/kube-router/pkg/controllers/routing/network_routes_controller.go:139 +0x147
> created by github.com/cloudnativelabs/kube-router/pkg/cmd.(*KubeRouter).Run
> 	/src/github.com/cloudnativelabs/kube-router/pkg/cmd/kube-router.go:156 +0xcbd

Tested on version **v0.2.1**, **v0.2.2**, **v0.2.3**, and a custom build that was built from master at the time of this issue. System setup is running **Kubernetes v1.13.0** on **CoreOS 1911.4.0** with kernel **4.14.81** with a kube-router configuration as follows.

> args:
>    - "--run-router=true"
>   - "--run-firewall=false"
>   - "--run-service-proxy=false"
>   - "--enable-cni=false"
>   - "--enable-ibgp=false"
>    - "--enable-overlay=false"
>   - "--advertise-cluster-ip=true"
>   - "--advertise-external-ip=true"
>   - "--advertise-loadbalancer-ip=true"
>   - "--enable-pod-egress=false"
>   - "--advertise-pod-cidr=true"
>   - "--nodes-full-mesh=false"
>   - "--bgp-graceful-restart=true"
>   - "--router-id=$(POD_IP)"
>   - "--v=5"
>   - "--cluster-asn=65000"

Note: kube-router.io/bgp-local-addresses, kube-router.io/peer-router-asns, and  kube-router.io/peer-router-ips annotations are being provided per node

I'm not sure what information you may need so please feel free to ask and I'm happy to provide anything you may need.

**Comment by murali-reddy** [comment]
Thanks for reporting. Let me try to reproduce. Do you suspect any thing @uablrek?

**Comment by uablrek** [comment]
@murali-reddy 

No, not really. But the ipv6 adaptation so far is limited to `--enable-cni=true` and the rest `=false` so this is really untested territory ipv6-wise.

A wild guess is that `kube-router` tries to add ipv6 stuff with (ipv4) `iptables` instead of `ip6tables` and the crash is caused by some unhandled error case. And there will be prenty of errors.

**Comment by davidjbrady** [comment]
@murali-reddy @uablrek

Full disclosure: I'm not a developer and come from a systems integration background so my findings may be naive. With that said... 

I looked into the "**iptables vs ip6tables**" theory and found the kube-router code already makes use of the `coreos/go-iptables` `NewWithProtocol` function which determines if the `iptables` or `ip6tables` command should be executed. In addition the protocol being passed into this function is being evaluated and doesn't appear to be hard coded to ipv4 in any manner.

I haven't exactly added anything here and not sure where this leaves us, but thought I'd share my findings.

**Comment by mk01** [comment]
there is ip6tables binary missing in the builds and code is not having check for return err value when creating iptables cmd handler. you can rebuild from official images
```
FROM cloudnativelabs/kube-router:v0.2.3
RUN apk add ip6tables && rm -rf /var/cache/apk/*
```

**Comment by murali-reddy** [comment]
@mk01 thanks for sharing this.

Does it make sense to fix it in current [Dockerfile](https://github.com/cloudnativelabs/kube-router/blob/master/Dockerfile)? Care to raise a PR?

**Comment by mk01** [comment]
@murali-reddy 

https://github.com/cloudnativelabs/kube-router/pull/631

**Comment by aauren** [comment]
This should be closed via #631. Please re-open if not.
